src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -146,6 +146,16 @@ def create_role_assignment(cmd, role, assignee=None, assignee_object_id=None, re
     if assignee_principal_type and not assignee_object_id:
         raise CLIError('usage error: --assignee-object-id GUID [--assignee-principal-type]')
 
+    # If assignee object id is provided without assignee principal type, try to get principle type from graph
+    if assignee_object_id and not assignee_principal_type:
+        graph_client = _graph_client_factory(cmd.cli_ctx)
+        try:
+            assignee_object_result = _get_object_stubs(graph_client, [assignee_object_id])
+            if assignee_object_result:
+                assignee_principal_type = assignee_object_result[0].object_type
+        except (CloudError, GraphErrorException):
+            pass
```

Comment:
Please also make sure to test the scenario where the logged-in account doesn't have permission to query Graph API and gets `403` which is more frequent than receiving a `404`.

Reply:
Perhaps we should warn the user when graph query fails? as server may reject `"principalType": none` in the future.

Reply:
`403` error example provided in PR description. Also added warning message 😊